### DISCUSSION TITLE
How many megabytes should we set the size of the memory?

### DISCUSSION THREAD

**Discussion by gtxzsxxk** [gtxzsxxk/duo-buildroot#8] [Q&A]
As we know, the cv1800b chip has a ION which will take some memory and let the free memory in linux become 35MiB. However, if we decide not to use the video decode, we could have all the free 64MiB memory. Which value should we set as default? Do you have any ideas?

**Comment by shodan8192** [comment]
There's no good default value, it depends purely on use case. Ideally, there should be a on/off switch for ION, that do two things :

- generates memmaps (fsbl, uboot) accordingly of use ION or not
- enables/disables ION in kernel

It can be tricky to achieve that, though.

**Reply by gtxzsxxk** [reply]
What about giving two defconfigs, such as `milkv_duo_ion` or `milkv_duo_64mib`? And their names here are just my examples. If this is a good plan in deed, we would re-consider the names.

**Reply by shodan8192** [reply]
Sure, if it's possible by having two configs it's a good idea

**Reply by gtxzsxxk** [reply]
This seems possible. See the `buildroot/configs`, I find the followings:

```shell
aarch64_efi_defconfig                     imx6ulpico_defconfig                            qemu_mips64_malta_defconfig
acmesystems_acqua_a5_256mb_defconfig      imx7dpico_defconfig                             qemu_mips64r6el_malta_defconfig
acmesystems_acqua_a5_512mb_defconfig      imx7d-sdb_defconfig                             qemu_mips64r6_malta_defconfig
acmesystems_aria_g25_128mb_defconfig      imx8mmpico_defconfig                            qemu_nios2_10m50_defconfig
acmesystems_aria_g25_256mb_defconfig      imx8mn_bsh_smm_s2_defconfig                     qemu_or1k_defconfig
acmesystems_arietta_g25_128mb_defconfig   imx8mn_bsh_smm_s2_pro_defconfig                 qemu_ppc64_e5500_defconfig
```

Maybe we should consider rename our defconfigs as `acmesystems_acqua_a5_***mb_defconfig` did.

**Reply by shodan8192** [reply]
Ok, kernel and uboot configs can be selected from defconfigs, but what about fsbl ? Or maybe its memmap is not relevant really ?

**Reply by shodan8192** [reply]
I've checked it, and memmap in fsbl also require to be altered. There are references to values from memmap in `bl2_main.c` that changes if ION memory is set to 0, look : [bl2_main.c#L20-L25](https://github.com/gtxzsxxk/milkv-duo-firmware-fsbl/blob/4367d680c81e8ba2c9d48eb07cd8ba1e1c25d9b9/plat/cv180x/bl2/bl2_main.c#L20-L25)

**Reply by gtxzsxxk** [reply]
Bad news. It seems like I had messed them up. Maybe I need to spend some time to think in which way we can dynamically alter the ION size in accord with the buildroot config.
